base/compiler/optimize.jl

```diff
@@ -707,9 +706,10 @@ function ((; sv)::ScanStmt)(inst::Instruction, idx::Int, lstmt::Int, bb::Int)
 end
 
 function check_inconsistentcy!(sv::PostOptAnalysisState, scanner::BBScanner)
-    scan!(ScanStmt(sv), scanner, true)
+    scan!(ScanStmt(sv), scanner, false)
```

Comment:
@Keno, I'm trying to understand the rationale behind the two-phase approach in `scan!` within `ipo_dataflow_analysis!`. Currently we first run `scan!` with the `forward_only` flag enabled, and in a case when the scan isn't completed (meaning control backedges are found), we then run it again with `forward_only` disabled. I'm concerned about the potential redundancy in this approach. Wouldn't it be more efficient to perform a full scan just once (and perform the additional inconsistency-check if necessary)?

Reply:
I don't think we run it again, we just continue running it with the same state.

Reply:
Ah, I got it. Thanks.